docs/technical-overview.md

```diff
@@ -0,0 +1,64 @@
+[Back to overview](./README.md)
+
+# Technical Overview
+
+The Plugin Checker works by executing checks against a single plugin. Each check will test for a specific issue and raise either an error or a warning depending on the severity.
+
+The Plugin Checker performs these checks by following the process below:
+
+1. Determine which checks to run based on the request.
+2. Execute necessary check preparations to set up the environment.
+3. Execute a comprehensive set of checks against the plugin.
+4. Ensure all errors and warnings raised during the checks are reported.
+
+
+## Checks
+Checks are individual PHP classes designed to test a plugin for a specific problem when executed.
+
+Each check contains a `run()` method that accepts an instance of the `Check_Result` class. This `Check_Result` instance is used to collect and report any errors or warnings raised by the check.
+
+The Plugin Checker currently supports two types of checks: static and runtime checks.
+
+### Static Checks
+
+Static checks are used to perform tests against the codebase without running any code, similar to other static analysis tools like PHP Code Sniffer.
```

Comment:
```suggestion
Static checks are used to perform tests against the codebase without running any code within the plugin being checked, similar to other static analysis tools like PHP Code Sniffer.
```

I know this may seem like a nit pick suggestion, but reading it prior when I read "perform tests against the codebase without running any code" it sounded like the plugin checker itself did not run any code, but of course it does :) I just added this to clarify that the plugin being checked has no code executed.